editor/plugins/animation_state_machine_editor.cpp

```diff
@@ -1743,7 +1966,125 @@ AnimationNodeStateMachineEditor::AnimationNodeStateMachineEditor() {
+	h_sep->set_anchors_and_offsets_preset(PRESET_TOP_WIDE);
+	h_sep->set_custom_minimum_size(Size2(0, 12));
+
+	visibility_reset_container = memnew(HBoxContainer);
+	sp_vertical_partition->add_child(visibility_reset_container);
+	visibility_reset_container->set_h_size_flags(SIZE_EXPAND_FILL);
+
+	show_all_ng_button = memnew(Button);
+	visibility_reset_container->add_child(show_all_ng_button);
+	show_all_ng_button->set_custom_minimum_size(Size2(50, 35));
+	show_all_ng_button->connect(SceneStringName(pressed), callable_mp(this, &AnimationNodeStateMachineEditor::_show_all_ng));
+	show_all_ng_button->set_expand_icon(true);
+	show_all_ng_button->set_icon_alignment(HORIZONTAL_ALIGNMENT_CENTER);
+	show_all_ng_button->set_h_size_flags(SIZE_EXPAND_FILL);
+
+	hide_all_ng_button = memnew(Button);
+	visibility_reset_container->add_child(hide_all_ng_button);
+	hide_all_ng_button->set_custom_minimum_size(Size2(50, 35));
+	hide_all_ng_button->connect(SceneStringName(pressed), callable_mp(this, &AnimationNodeStateMachineEditor::_hide_all_ng));
+	hide_all_ng_button->set_expand_icon(true);
+	hide_all_ng_button->set_icon_alignment(HORIZONTAL_ALIGNMENT_CENTER);
+	hide_all_ng_button->set_h_size_flags(SIZE_EXPAND_FILL);
+
+	for (int i = 0; 12 > i; i++) {
+		ng_hor_containers[i] = memnew(HBoxContainer);
+		sp_vertical_partition->add_child(ng_hor_containers[i]);
+		ng_containers[i] = memnew(HBoxContainer);
+		ng_hor_containers[i]->add_child(ng_containers[i]);
+		ng_containers[i]->set_h_size_flags(SIZE_EXPAND);
+		ng_name_line[i] = memnew(LineEdit);
+		ng_containers[i]->add_child(ng_name_line[i]);
+		ng_name_line[i]->set_custom_minimum_size(Size2(100, 30));
+		ng_name_line[i]->set_horizontal_alignment(HORIZONTAL_ALIGNMENT_CENTER);
+		ng_name_line[i]->set_visible(false);
+
+		show_ng_buttons[i] = memnew(Button);
+		show_ng_buttons[i]->set_custom_minimum_size(Size2(25, 25));
+		show_ng_buttons[i]->connect(SceneStringName(pressed), callable_mp(this, &AnimationNodeStateMachineEditor::_show_ng_pressed));
+
+		hide_ng_buttons[i] = memnew(Button);
+		hide_ng_buttons[i]->set_custom_minimum_size(Size2(25, 25));
+		hide_ng_buttons[i]->connect(SceneStringName(pressed), callable_mp(this, &AnimationNodeStateMachineEditor::_hide_ng_pressed));
+
+		ng_containers[i]->add_child(show_ng_buttons[i]);
+		ng_containers[i]->add_child(hide_ng_buttons[i]);
+
+		ng_select_checkboxes[i] = memnew(CheckBox);
+		ng_labels[i] = memnew(Label);
+
+		ng_containers[i]->add_child(ng_select_checkboxes[i]);
+		ng_containers[i]->add_child(ng_labels[i]);
+		ng_select_checkboxes[i]->set_visible(false);
+		ng_labels[i]->set_visible(false);
+		ng_select_checkboxes[i]->connect(SceneStringName(pressed), callable_mp(this, &AnimationNodeStateMachineEditor::_checkbox_pressed));
+	}
+
+	spacer = memnew(HBoxContainer);
+	spacer->set_v_size_flags(SIZE_EXPAND);
+	sp_vertical_partition->add_child(spacer);
+
+	ng_selection_menu_button = memnew(Button);
+	ng_selection_menu_button->set_text("Select Node Groups");
```

Comment:
```suggestion
	ng_selection_menu_button->set_text(TTR("Select Node Groups"));
```
Need translation for all the ones you've added